packages/jwt/src/JwtBase.ts

```diff
@@ -0,0 +1,100 @@
+import jsonwebtoken from 'jsonwebtoken';
+import { utils } from 'ethers';
+import base64url from 'base64url';
+import { pubToPem, KeyType } from '@ew-did-registry/keys';
+import { computePublicKey } from 'ethers/lib/utils';
+import { JwtVerifyOptions, Algorithms } from './types';
+import { JwtVerificationFailed } from './JwtVerificationFailed';
+
+const {
+  arrayify, keccak256, hashMessage, recoverPublicKey,
+} = utils;
+
+/**
+ * @description Any JWT implementation verifies any JWT
+ */
+export class JwtBase {
+  /**
+  * If the signature is correct, method returns decoded JWT payload
+  *
+  * @example
+  * ```typescript
+  * const issuerKeys = new Keys();
+  * const issuer = new JWT(issuerKeys);
+  * const verifier = new JWT(new Keys());
+  * const payload = {claim: 'test'};
+  *
+  * const token = await issuer.sign(payload); // signing algorithm is determined by issuer's signer
+  * expect(verifier.verify(token), issuerKeys.publicKey, ['ES256K']).to.eql(payload);
+  * ```
+  * @param {string} token
+  * @param {string} publicKey - public key in hexadecimal format in compressed or uncompressed form
+  */
+  verify(
+    token: string,
+    publicKey: string,
+    { algorithms = [Algorithms.EIP191] }: JwtVerifyOptions = {},
+  ): unknown {
+    if (algorithms.includes(Algorithms.EIP191)) {
+      try {
+        return this.verifyES256K(token, publicKey);
```

Comment:
I think it would be better if the `verify` function also reflected the Algo name
```suggestion
        return this.verifyEIP191(token, publicKey);
```